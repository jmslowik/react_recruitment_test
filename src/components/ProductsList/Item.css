.row {
  padding: 10px;
  border-bottom: 1px solid silver;
  list-style: none;
  display: flex;
}

.after {
  margin-left: auto;
}

.price {
  margin-left: auto;
}

.item {
  width: 60%;
  display: flex;
}