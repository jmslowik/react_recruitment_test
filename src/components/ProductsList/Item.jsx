import React, { useContext } from 'react';
import LocaleContext from '@/contexts/LocaleContext';
import './Item.css';

const Item = ({
  name, price = 0, after,
}) => {
  const { locale, currency } = useContext(LocaleContext);

  return (
    <li className="row">
      <div className="item">
        {`${name}:`}
        <span className="price">
          {price.toLocaleString(locale, { style: 'currency', currency })}
        </span>

      </div>
      <div className="after">{after}</div>
    </li>
  );
};

export default Item;
