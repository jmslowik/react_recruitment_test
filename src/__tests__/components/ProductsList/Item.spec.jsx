import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import Item from '@/components/ProductsList/Item';

describe('Item', () => {
  const After = () => <div className="after-prop">after</div>;
  const name = 'name';
  const price = 11.1;

  it('should render without throwing an error', () => {
    const wrapper = shallow(<Item />);
    expect(wrapper.is('.row')).toBeTruthy();
  });

  it('should contain name under .item', () => {
    const wrapper = shallow(<Item name={name} />);
    expect(wrapper.find('.item').text()).toMatch(/^name:/);
  });

  it('should contain price under .price', () => {
    const wrapper = shallow(<Item price={11} />);
    expect(wrapper.find('.price').text()).toMatch(/^11/);
  });

  it('should format price correctly', () => {
    const wrapper = shallow(<Item price={price} />);
    expect(wrapper.find('.price').text()).toEqual('11,10 zł');
  });

  it('should inject after conponent under .after', () => {
    const wrapper = mount(<Item after={<After />} />);
    expect(wrapper.find('.after').first().contains(<After />)).toBeTruthy();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<Item name={name} price={price} after={<After />} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
