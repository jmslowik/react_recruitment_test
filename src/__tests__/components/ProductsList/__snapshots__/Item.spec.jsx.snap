// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render correctly 1`] = `
<li
  className="row"
>
  <div
    className="item"
  >
    name:
    <span
      className="price"
    >
      11,10 zł
    </span>
  </div>
  <div
    className="after"
  >
    <div
      className="after-prop"
    >
      after
    </div>
  </div>
</li>
`;
